ios/xcode/MGLKit/MGLLayer.mm

```diff
@@ -710,6 +711,18 @@ - (void)ensureSurfaceCreated
             red = green = blue = alpha = 8;
             colorSpace                 = EGL_GL_COLORSPACE_SRGB_KHR;
             break;
+        case MGLDrawableColorFormatRGBA16:
+            red = green = blue = alpha = 16;
```

Comment:
Using `16` causes `Failed to call eglChooseConfig()` exception. I think it is probably unnecessary since we are setting the metal pixel format and color space correctly.

Reply:
For a new config to be accepted (such as RGBA 16 bits), it must be added to the supported list inside `DisplayMtl::generateConfigs()`

Reply:
Oh okay, I see. That function is a bit confusing to me, but I will try to figure out how to add the 15bit formats in there.

Reply:
I was able to make this work. But I still don't understand what effect this has, as opposed to fbo format and texture formats used.

Reply:
Currently SurfaceMtl assumes users always create a window surface with 8 bits format (this is actually guaranteed by `DisplayMtl::generateConfigs()`). Hence in this code, there wasn't any color channels' configured bits size's verification, thus it always chooses `MTLPixelFormatBGRA8Unorm ` metal format for default colorspace:
https://github.com/kakashidinho/metalangle/blob/78017f07e5adee0692506c94cdb299d92c854ad3/src/libANGLE/renderer/metal/SurfaceMtl.mm#L219

If you add a new 16 bits config with default colorspace, the code should examine the color channels's configured bits size inside `egl::SurfaceState` parameter and choose appropriate metal format such as `MTLPixelFormatBGRA8Unorm` or `MTLPixelFormatRGBA16Float` similar to how depth & stencil bits configuration is handled:
https://github.com/kakashidinho/metalangle/blob/78017f07e5adee0692506c94cdb299d92c854ad3/src/libANGLE/renderer/metal/SurfaceMtl.mm#L228
https://github.com/kakashidinho/metalangle/blob/78017f07e5adee0692506c94cdb299d92c854ad3/src/libANGLE/renderer/metal/SurfaceMtl.mm#L232

However, I just realized you have no use for RGBA16 with default color space, maybe we can just omit  `MGLDrawableColorFormatRGBA16` and only keep its BT.2020 variant. Thus further hassles would be avoided.

Reply:
Thanks! I will reduce scope to BT2020 variant only as you suggested.